Bottom: 19fd94debab3fbd2372bf150ad89758ae2ef22e7
Top:    5f0c0fd0431f3f6102b7c6b0510a960075d81b23
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-10 12:19:58 -0500

Always update configuration on startup

The entrypoint script will always copy updated certificates and incorporate
updated configuration on container startup.


---
diff --git a/docker/docker-exim4/docker-entrypoint.sh b/docker/docker-exim4/docker-entrypoint.sh
index 8d9d35cb..3c3e0a26 100755
--- a/docker/docker-exim4/docker-entrypoint.sh
+++ b/docker/docker-exim4/docker-entrypoint.sh
@@ -20,27 +20,25 @@ chown -R root:mail /var/mail
 
 case ${1} in
     exim4)
-        if   [[ -f ${CONFIG_PATH}/update-exim4.conf.conf ]] \
-          && [[ -f ${CONFIG_PATH}/exim4.conf.template ]] \
-          && [[ ! -f ${CONFIG_PATH}/exim4.conf ]]
-        then
-            # set the hostname
-            echo "${HOSTNAME}" > /etc/mailname
-            # copy certificate
-            [ -e /etc/ssl/private/exim.crt ] \
-                && rm -f ${CONFIG_PATH}/exim.crt \
-                && cp $(realpath /etc/ssl/private/exim.crt) ${CONFIG_PATH}/exim.crt \
-                && chmod 640 ${CONFIG_PATH}/exim.crt \
-                && chown root:Debian-exim ${CONFIG_PATH}/exim.crt
-            # copy key
-            [ -e /etc/ssl/private/exim.key ] \
-                && rm -f ${CONFIG_PATH}/exim.key \
-                && cp $(realpath /etc/ssl/private/exim.key) ${CONFIG_PATH}/exim.key \
-                && chmod 640 ${CONFIG_PATH}/exim.key \
-                && chown root:Debian-exim ${CONFIG_PATH}/exim.key
-            # run the debian utility to update/generate the configuration
-            $(which update-exim4.conf) --verbose
-        fi
+        # set the hostname
+        echo "${HOSTNAME}" > /etc/mailname
+
+        # copy certificate
+        [ -e /etc/ssl/private/exim.crt ] \
+            && rm -f ${CONFIG_PATH}/exim.crt \
+            && cp $(realpath /etc/ssl/private/exim.crt) ${CONFIG_PATH}/exim.crt \
+            && chmod 640 ${CONFIG_PATH}/exim.crt \
+            && chown root:Debian-exim ${CONFIG_PATH}/exim.crt
+
+        # copy key
+        [ -e /etc/ssl/private/exim.key ] \
+            && rm -f ${CONFIG_PATH}/exim.key \
+            && cp $(realpath /etc/ssl/private/exim.key) ${CONFIG_PATH}/exim.key \
+            && chmod 640 ${CONFIG_PATH}/exim.key \
+            && chown root:Debian-exim ${CONFIG_PATH}/exim.key
+
+        # run the debian utility to update/generate the configuration
+        $(which update-exim4.conf) --verbose
         
         # Check configuration
         if ! /usr/sbin/exim4 -bV > /dev/null ; then
